src/lib.rs

```diff
@@ -62,30 +63,46 @@ impl Decoder {
             }
 
             Decoder {
-                dec: dec.assume_init(),
+                dec: ptr::NonNull::new(dec.assume_init()).unwrap(),
+                pending_data: None,
             }
         }
     }
 
-    pub fn flush(&self) {
+    pub fn flush(&mut self) {
```

Comment:
This should've been `&mut` from the beginning as it modifies the decoder state

Reply:
Sounds right.